lib/perl/Genome/Model/Build/ReferenceSequence/AnnotationIndex.pm

```diff
@@ -79,227 +41,33 @@ sub __display_name__ {
-
-    $class->debug_message(sprintf("Resolved aligner class %s, making sure it's real and can be loaded.", $aligner_class));
-    unless ($aligner_class->class) {
-        $class->error_message(sprintf("Failed to load aligner class (%s).", $aligner_class));
-        return;
-    }
-
-    if ($class->aligner_requires_param_masking($p{aligner_name})) {
-        $p{aligner_params} = undef;
-    }
-
-    my $self = $class->SUPER::create(%p);
-    return unless $self;
-    $self->aligner_class_name($aligner_class);
-
-    $self->debug_message("Prepare staging directories...");
-    unless ($self->_prepare_staging_directory) {
-        $self->error_message("Failed to prepare working directory");
-        return;
-    }
-
-    unless ($self->_prepare_annotation_index) {
-        $self->error_message("Failed to prepare annotation index!");
-        return;
-    }
-
-    unless ($self->check_dependencies()) {
-        $self->error_message("Failed to create AnnotationIndex objects for dependencies");
-        return;
-    }
-
-    return $self;
-}
-
-sub check_dependencies {
+sub generate_dependencies_as_needed {
     my $self = shift;
 
-    my %params = (
-        aligner_name => $self->aligner_name,
-        aligner_params => $self->aligner_params,
-        aligner_version => $self->aligner_version,
-    );
-
     # if the reference is a compound reference
     if ($self->reference_build->append_to and $self->_supports_multiple_reference) {
         die('Compound references are not currently supported in '. __PACKAGE__);
-        for my $b ($self->reference_build->append_to) { # (append_to is_many)
-            $params{reference_build} = $b;
-            $self->debug_message("Creating AlignmentIndex for build dependency " . $b->name);
-            my $result = Genome::Model::Build::ReferenceSequence::AlignerIndex->get_or_create(%params);
-            unless($result) {
-                $self->error_message("Failed to create AlignmentIndex for dependency " . $b->name);
-                return;
-            }
-            unless ($result->check_dependencies()) {
-                $self->error_message("Failed while checking dependencies of " . $b->name);
-                return;
-            }
-        }
     }
+
     return 1;
 }
 
-sub _prepare_annotation_index {
+sub _prepare_index {
     my $self = shift;
-
-    my $reference_fasta_file;
-    if ($self->_supports_multiple_reference) {
-        $reference_fasta_file = $self->reference_build->primary_consensus_path('fa');
-    } else {
-        $reference_fasta_file = $self->reference_build->full_consensus_path('fa');
-    }
-
-    unless (-s $reference_fasta_file) {
-        $self->error_message(sprintf("Reference fasta file %s does not exist", $reference_fasta_file));
-        return;
-    }
-
-    $self->debug_message(sprintf("Confirmed non-zero reference fasta file is %s", $reference_fasta_file));
+    my $reference_fasta_file = shift;
```

Comment:
$reference_fasta_file is not actually used in this method, although it is part of the interface.  Is it better to shift it explicitly since it is part of the interface, or ignore it since it isn't used?